pkg/repository/local_git.go

```diff
@@ -345,24 +363,104 @@ func (lg *LocalGit) GetDirContentSHA(ctx context.Context, gitPath, dirPath strin
 
 	filteredLS := strings.Join(filteredLines, "\n")
 
-	hashObjectCmdArgs := []string{"--no-optional-locks", "hash-object", "--stdin"}
-	output, err := lg.exec.RunPipeCommands(ctx, gitPath, "echo", []string{filteredLS}, lg.gitPath, hashObjectCmdArgs)
+	// As the list of files might be too long, we write it to a file and calculate the hash of the file
+	f, err := lg.fs.Create(filepath.Join(config.GetOktetoHome(), fmt.Sprintf("git-untracked-%s-%d", filepath.Base(dirPath), time.Now().UnixNano())))
```

Comment:
I don't really know if I like this. I think it should be preferable to write into a temple instead of okteto home. 

If we have a problem removing it if we do it in a temp file when the computer restarts it will be deleted, while if we do this in the okteto home we could be increasing the folder

Reply:
We have this problem in other parts of the CLI like the custom kubeconfig for the deploys. We should unify and standardize how we do it, but I don't know if we should do it as part of this PR or not